### PR TITLE
fix: prevent swallowing of input validation errors

### PR DESCRIPTION
**Pull request by Aenimus**
#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: testapps/default/.wundergraph/wundergraph.config.ts [hunk 1]
```diff
@@ -108,12 +108,12 @@ configureWunderGraphApplication({
 						// change this before deploying to production to the actual domain where you're deploying your app
 						'http://localhost:3000',
 				  ]
-				: ['http://localhost:3000'],
+				: ['http://localhost:3000', 'http://localhost:9991'],
```
Comment (Aenimus) [inline testapps/default/.wundergraph/wundergraph.config.ts:111]: I will revert this.

**Review by fiam — APPROVED** [review]
LGTM. Excited to get better error messages!

**Review by Pagebakers — CHANGES_REQUESTED** [review]
Looks great!

One thing though. In our SWR and React Query clients we use GraphqlResponseError as the error type. I believe we should rewrite this to ClientResponseError instead. So useQuery and useMutation return the correct error type.

File: packages/react-query/src/types.ts [hunk 2]
```diff
@@ -14,6 +13,7 @@ import {
 	UseQueryResult,
 	UseMutationResult,
 } from '@tanstack/react-query';
+import { ClientResponseError } from '@wundergraph/sdk/dist/client/ClientResponseError';
```
Comment (spetrunin) [inline packages/react-query/src/types.ts:16]: I believe import should not be from dist, but from package itself

```
@wundergraph/sdk/client/ClientResponseError'
```

maybe you will need to export it ether from client to been able to to this
```
import { ClientResponseError } from '@wundergraph/sdk/dist/client';
```

**Review by Pagebakers — APPROVED** [review]
Nice work!

**Pull request merged**